lib/Doctrine/ORM/EntityManager.php

```diff
@@ -479,8 +474,13 @@ public function getReference($entityName, $id)
         foreach ($id as $i => $value) {
             $scalarId[$i] = $value;
 
-            if (is_object($value) && $this->metadataFactory->hasMetadataFor(StaticClassNameConverter::getClass($value))) {
-                $scalarId[$i] = $this->unitOfWork->getSingleIdentifierValue($value);
+            if (is_object($value)) {
+                try{
+                    $this->metadatas->get(StaticClassNameConverter::getClass($value));
+                    $scalarId[$i] = $this->unitOfWork->getSingleIdentifierValue($value);
+                } catch (\Exception $e) {
```

Comment:
Likely wrong here - why did you introduce these try-catches? I know you are still playing around with the patch: just wondering why this got added to make things work